arviz/stats/stats.py

```diff
@@ -1331,52 +1353,57 @@ def summary(
-            _make_ufunc(_multichain_statistics, n_output=5, ravel=False),
-            dataset,
-            input_core_dims=(("chain", "draw"),),
-            output_core_dims=tuple([] for _ in range(5)),
-        )
+        if stat_focus == "mean":
+            mcse_mean, mcse_sd, ess_bulk, ess_tail, r_hat = xr.apply_ufunc(
+                _make_ufunc(_multichain_statistics, n_output=5, ravel=False),
+                dataset,
+                input_core_dims=(("chain", "draw"),),
+                output_core_dims=tuple([] for _ in range(5)),
+            )
+        elif stat_focus == "median":
+            mcse_median, ess_median, ess_tail, r_hat = xr.apply_ufunc(
+                _make_ufunc(_multichain_statistics, n_output=4, ravel=False),
+                dataset,
+                kwargs=dict(focus="median"),
+                input_core_dims=(("chain", "draw"),),
+                output_core_dims=tuple([] for _ in range(4)),
+            )
 
     # Combine metrics
-    metrics: List[xr.Dataset] = []
-    metric_names: List[str] = []
-    if extend:
-        metrics_: Tuple[xr.Dataset, ...]
-        metrics_names_: Tuple[str, ...] = (
-            "mean",
-            "sd",
-            f"hdi_{100 * alpha / 2:g}%",
-            f"hdi_{100 * (1 - alpha / 2):g}%",
-            "mcse_mean",
-            "mcse_sd",
-            "ess_bulk",
-            "ess_tail",
-            "r_hat",
-        )
-        if kind == "all":
-            metrics_ = (
-                mean,
-                sd,
-                hdi_lower,
-                hdi_higher,
+    if extend and kind in ["all", "diagnostics"]:
+        if stat_focus == "mean":
+            metrics_: Tuple[xr.Dataset, ...] = (
                 mcse_mean,
                 mcse_sd,
                 ess_bulk,
                 ess_tail,
                 r_hat,
             )
-        elif kind == "stats":
-            metrics_ = (mean, sd, hdi_lower, hdi_higher)
-            metrics_names_ = metrics_names_[:4]
-        elif kind == "diagnostics":
-            metrics_ = (mcse_mean, mcse_sd, ess_bulk, ess_tail, r_hat)
-            metrics_names_ = metrics_names_[4:]
+            metrics_names_: Tuple[str, ...] = (
+                "mcse_mean",
+                "mcse_sd",
+                "ess_bulk",
+                "ess_tail",
+                "r_hat",
+            )
+        elif stat_focus == "median":
+            metrics_: Tuple[xr.Dataset, ...] = (
+                mcse_median,
+                ess_median,
+                ess_tail,
+                r_hat,
+            )
+            metrics_names_: Tuple[str, ...] = (
+                "mcse_median",
+                "ess_median",
+                "ess_tail",
+                "r_hat",
+            )
         metrics.extend(metrics_)
         metric_names.extend(metrics_names_)
 
     if circ_var_names:
-        if kind != "diagnostics":
+        if kind != "diagnostics" and stat_focus == "mean":
```

Comment:
This should work in both cases. We only need to be careful to replace the right positions in `metrics`. I don't think there is a clear definition of a circular median (nor implementation in scipy for example) so it is fine for circular variables to report the mean instead. We only need to document that properly.